horovod/run/common/service/task_service.py

```diff
@@ -194,14 +194,31 @@ def wait_for_initial_registration(self, timeout):
             self._wait_cond.release()
 
     def wait_for_command_start(self, timeout=None):
+        """Waits for a command to start.
```

Comment:
The overloaded behavior is a little tricky to follow.  I would consider breaking this out into two methods: one that raises an exception and takes a Timeout object, and another that takes a numerical value and returns the state at the end of the timeout.

Reply:
You are right, this is horrible. I have split it up into two methods. Thanks!